docs/API.md

```diff
@@ -162,16 +164,19 @@ A simple representation of the store's lifecycle:
   - Note that there are some "custom" built-in events at this step:
     - `ACCOUNT_TRIGGER`: triggered whenever the currently connected account changes
 
-However, with `options.init` and `options.externals`, the lifecycle becomes a bit more complicated:
+If `options.init` and `options.externals` are given, the lifecycle becomes a bit more complicated:
 
 1. Obtain the initial "committed" state, as before
 1. If `options.init` is available, feed the initial state into `options.init`. Use the returned state from `options.init` as the new current state.
-1. Fetch past events from the contract and any given `options.externals` contracts, reducing new state from found events. Note that new events from these contracts are **not** fetched until all past events have been found.
-1. Cache the state as "committed" state, as before
+1. Fetch past events from the application contract and any given `options.externals` contracts, reducing new state from found events. Note that new events from both the application contract and external contracts are **not** fetched until all past events have been found.
+1. Cache the state at the end of the initial sync as "committed" state, as before
 1. Subscribe to new events from the contract and any given `options.externals` contracts, reducing new state based on the incoming events.
 
+If the application emits its own `trigger`s (see [`emitTrigger`](#emittrigger)), the triggers will appear in the reducer only once the syncing phase for past events ends. Any triggers emitted before this will be ignored. Triggers are implicitly transformed into the web3 event schema of `event` (string) and `returnValues` (object).
```

Comment:
> The triggers will appear in the reducer only once the syncing phase for past events ends. Any triggers emitted before this will be ignored

This is a pretty big limitation in the current implementation, given that the initial sync can be quite lengthy. One way to get around this is to only emit triggers from the frontend once you've detected the sync period is over (`SYNC_STATUS_SYNCED`), but this workaround will not work in all cases (e.g. when trying to access an external API before the initial sync is over) and is somewhat silly.

I have a change to `store()` that will fix this, so triggers will be emitted during the past events' sync phase as well, but this is a more sensitive change and I'd like to test it before pushing.

Reply:
I think this is fine for now, since in most of our use cases you'll want to allow the past events to be synced before interacting with the application anyways.

Reply:
Confirmed a96b0f4 works for propagating triggers during the past events phase.

Due to the way the events are streamed, all past events from a contract are handled "in order" and don't give up control to a trigger that is queued. This is most noticeable if you have a trigger emitting on a timed interval, where on the past events being fetched, the timed trigger will act as if it's been paused (in reality it's just getting queued and will all be flooded into the reducer when the past events have finished reducing).